task16/index.js

```diff
@@ -0,0 +1,108 @@
+    loadingPlace.id = "loadingPlace";
+    tabWrapper.appendChild(loadingPlace);
+}
+
+const removeLoading = () => document.getElementById("loadingPlace").remove();
+
+const getFetchData = async () => {
+    try{
+        const respons = await fetch(URL);
+        if(!respons.ok){
+            throw new Error('ただいまサーバー側で通信が壊れています')
+        };
+        const json = await respons.json();
+        return json;
+    }
+    catch(error) {
+        console.error(error);
+    }
+};
+
+const init = async () => {
+    addLoading();
+    try {
+        const value = await getFetchData();
+        renderTabMenu(value);
+        renderTabContainer(value);
+        // tabSwitch();
+    }
+    catch(error) {
+        console.error(error);
+        tab.textContent = "データを読み込めませんでした";
+    }
+    finally {
+        removeLoading();
+    }
+};
+init();
+
+const renderTabMenu = (value) => {
+    const tabMenuFragment = document.createDocumentFragment();
+    for(let i = 0; i < value.length; i++){
+        const menuList = createElementWithClassName('li', "menu-list");
+        menuList.id = `js-tabMenuList${i}`;
+        menuList.textContent = value[i].category;
+        tabMenuFragment.appendChild(menuList);
+    }
+    tab.appendChild(tabMenuFragment);
+    const addClassActive = document.getElementById("js-tabMenuList0");
+    addClassActive.classList.add("active");
+};
+
+const renderTabContainer = (value) => {
+    const tabContentsFragment = document.createDocumentFragment();
+    for(let i = 0; i < value.length; i++){
+        const newsWrapper = createElementWithClassName('div', "news-wrapper");
+        const newsTitlelist = createElementWithClassName('ul', "news-title-list");
+        const newsImg = createElementWithClassName('img', "news-img")
+        newsWrapper.id = `js-newsWrapper${i}`;
+        newsImg.src = value[i].img;
+
+        tabContentsFragment.appendChild(newsWrapper).appendChild(newsTitlelist).appendChild(getNewsTitle(value[i]));
+
+        tabContentsFragment.appendChild(newsWrapper).appendChild(newsImg);
+    }
+    const tabContainer = createElementWithClassName('div', "tab-container");
+    tabWrapper.appendChild(tabContainer).appendChild(tabContentsFragment);
+    const addClassShow = document.getElementById("js-newsWrapper0");
+    addClassShow.classList.add("show");
+};
+
+const getNewsTitle = ({article}) => {
+    const newsTitleFragment = document.createDocumentFragment();
+    for(let i = 0; i < article.length; i++){
+        const newsTitleItem = createElementWithClassName('li', "news-title-item");
+        const newsLink = createElementWithClassName('a', "news-link");
+        newsLink.textContent = article[i].title;
+        newsLink.href = "#";
+
+        newsTitleFragment.appendChild(newsTitleItem).appendChild(newsLink);
+    }
+    return newsTitleFragment;
+};
+
+    tab.addEventListener('click', (e) => {
+    const active = document.querySelector(".active");
```

Comment:
nits
I don't think the indentation is consistent.
Sorry if that was your intention.

Reply:
Sorry I wasn't coherent.
I will try to press this command last when writing code and will take care of it next time.
Thank you so much for even teaching me the command.
f740b9d